### ISSUE TITLE
feat: Implement output layer for vocabulary prediction

### ISSUE THREAD

**Issue by muuki2** [muuki2/makemore-clone#6]
Add the final linear layer mapping hidden activations to vocabulary logits.
